nixos/tests/zfs.nix

```diff
@@ -31,7 +31,7 @@ let
         boot.loader.systemd-boot.enable = true;
         boot.loader.timeout = 0;
         boot.loader.efi.canTouchEfiVariables = true;
-        networking.hostId = "deadbeef";
+        networking.machineId = "deadbeefdeadbeefdeadbeefdeadbeef";
```

Comment:
Let's instead add a separate NixOS test for this which checks that the `/etc/machine-id` contains the correct contents and also that `/etc/hostid` is derived in a constant way, i.e. that the algorithm which derives it doesn't change in the future.

Reply:
Because ZFS really has nothing to do with the machine ID.